### ISSUE TITLE
to_std is lossy

### ISSUE THREAD

**Issue by elbaro** [PoiScript/iso8601-duration#1]
```
let d = ...::parse("PT1H").unwrap();
println!("{:?}\n{:?}", d, d.to_std());

Duration { year: 0.0, month: 0.0, day: 0.0, hour: 1.0, minute: 0.0, second: 0.0 }
3599.99995904s
```

**Comment by rrichardson** [comment]
@elbaro - It might be a bit late, but I put in a PR to fix the to_std() issue.   
I can't guarantee that it won't ever happen, because floating point values are somewhat probabilistic, but it shouldn't happen under normal circumstances.

**Comment by elbaro** [comment]
Didn't know that iso8601 allows fractions in the place of hours or minutes.

Now I see how difficult this is - a small floating error in the number of days becomes non-negligible when it's multiplied by `24*60*60*1e9 ns/day`. Supporting common use cases would be fine.

**Comment by FrederickFrance** [comment]
Great
Same problem
I hope your PR will be merged

**Comment by rrichardson** [comment]
I'm starting to think that for practical applications, we should just ban the use of Month for anything other than parsing.

**Comment by PoiScript** [comment]
Sorry for the delayed response! I just released a 0.2 version to solve this long-exist problem:

1. `Duraion::to_std` now returns a `None` when `year` or `month` is used. 
2. To get an accurate duration, you can enable `chrono` feature and specify a staring date by call `Duration::to_chrono_at_datetime`:

```rust
use iso8601_duration::Duration;
use chrono::DateTime;

let one_month: Duration = "P1M".parse().unwrap();
let date = DateTime::parse_from_rfc3339("2000-02-01T00:00:00Z").unwrap();
assert_eq!(
    one_month.to_chrono_at_datetime(date).num_days(),
    29 // 2000 is a leap year
);
```
